servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLRequestHandler.kt

```diff
@@ -37,8 +38,6 @@ open class GraphQLRequestHandler(
      * Subscriptions require more specific server logic and will need to be handled separately.
      */
     open suspend fun executeRequest(request: GraphQLRequest, context: GraphQLContext? = null): GraphQLResponse<*> {
-        // We should generate a new registry for every request
-        val dataLoaderRegistry = dataLoaderRegistryFactory?.generate()
```

Comment:
i thought that data loader registry should be re-generated per request?

Reply:
From graphql-java: 

Per Request Data Loaders
If you are serving web requests then the data can be specific to the user requesting it. If you have user specific data then you will not want to cache data meant for user A to then later give it to user B in a subsequent request.

The scope of your DataLoader instances is important. You will want to create them per web request to ensure data is only cached within that web request and no more. It also ensures that a dispatch call only affects that graphql execution and no other.

DataLoaders by default act as caches. If they have seen a value before for a key then they will automatically return it in order to be efficient.

If your data can be shared across web requests then you might want to change the caching implementation of your data loaders so they share data via a caching layer say like memcached or redis.

You still create data loaders per request, however the caching layer will allow data sharing (if that’s suitable).

Reply:
So based off of that we may not actually want to use the new format and users shouldn't create data loaders as beans?

Reply:
This is where again I think there is no need for data loaders in our library but I understand some users want to use pre-built caching solutions.

Reply:
I'm thinking we could keep the KotlinDataLoader interface but change from val field to fun to generate the dataloader. We could then easily recreate the registry per request.

Reply:
And yeah we probably should keep the registry factory